resources-manager/src/AdaptivePruningManager.cpp

```diff
@@ -0,0 +1,81 @@
+#include "AdaptivePruningManager.hpp"
+#include "Bitmap.hpp"
+#include "ReplicaConfig.hpp"
+#include "concord.cmf.hpp"
+#include "SigManager.hpp"
+
+using namespace concord::performance;
+
+AdaptivePruningManager::AdaptivePruningManager(
+    const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
+    const std::chrono::duration<double, std::milli> &interval,
+    const std::shared_ptr<bftEngine::impl::IInternalBFTClient> &bftClient)
+    : repId(bftEngine::ReplicaConfig::instance().getreplicaId()),
+      resourceManager(resourceManager),
+      mode(LEGACY),
+      interval(interval),
+      bftClient(bftClient) {
+  start();
+}
+
+AdaptivePruningManager::~AdaptivePruningManager() { stop(); }
+
+void AdaptivePruningManager::notifyReplicas(const long double &rate, const uint64_t batchSize) {
+  concord::messages::ReconfigurationRequest rreq;
+
+  concord::messages::PruneTicksChangeRequest pruneRequest;
+  pruneRequest.sender_id = bftClient->getClientId();
+  pruneRequest.tick_period_seconds =
```

Comment:
in case the rate is lower than cluster size it will result in 0 rate.
the logic should be more sophisticated, to know how to "play" with the rate in order to support different bps options.

Reply:
Totally agree, but I think we should have it simple for now. Currently, the minimum supported tick per second is 1 and it can't go lower.